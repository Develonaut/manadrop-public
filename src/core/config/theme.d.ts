import { Theme } from "@material-ui/core/styles/createMuiTheme";
import { Palette } from "@material-ui/core/styles/createPalette";

declare module "@material-ui/core/colors/common" {
  interface CommonColors {
    gideon: string;
    ajani: string;
    karn: string;
    mana: string;
    rakdos: string;
    vraska: string;
    swamp: string;
    ral: string;
    liliana: string;
    chandra: string;
    plains: string;
    mountain: string;
    forest: string;
    nissa: string;
    island: string;
  }
}
declare module "@material-ui/core/styles/createMuiTheme" {
  // Allow configuration using `createMuiTheme`
  interface ThemeOptions {
    dimensions?: object;
    zIndexScale?: any;
  }
}

declare module "@material-ui/core/styles/createPalette" {
  interface TypeBackground {
    darkest: string;
    dark: string;
    light: string;
    lightest: string;
  }
}
