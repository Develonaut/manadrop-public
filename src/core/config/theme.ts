/// <reference types="./theme" />

import createMuiTheme, {
  ThemeOptions
} from "@material-ui/core/styles/createMuiTheme";

export const drawerWidth = 279;

const zIndexScale = {
  zIndex0: 0,
  zIndex1: 100,
  zIndex2: 200,
  zIndex3: 300,
  zIndex4: 400,
  zIndex5: 500,
  zIndex6: 600,
  zIndex7: 700,
  zIndex8: 800,
  zIndex9: 900,
  zIndex10: 1000
};

export const zIndex = {
  collectionDetailsBackground: zIndexScale.zIndex4,
  collectionCountBackground: zIndexScale.zIndex4,
  collectionDetailsForeground: zIndexScale.zIndex5,
  collectionCountForeground: zIndexScale.zIndex5,
  cardPopover: zIndexScale.zIndex5,
  cardCount: zIndexScale.zIndex5
};

export const colors = {
  gideon: "#fafefd",
  ajani: "#eae8fd",
  karn: "#9d96af",
  mana: "#281a3a",
  rakdos: "#33214a",
  vraska: "#3a2655",
  swamp: "#452f63",
  ral: "#9170a2",
  liliana: "#bc73ff",
  chandra: "#ffa000",
  plains: "#e5a52f",
  mountain: "#f35c6f",
  nissa: "#01c7c8",
  forest: "#09b8b9",
  island: "#4a40e8"
};

const backgrounds = {
  darkest: colors.mana,
  dark: colors.rakdos,
  light: colors.vraska,
  lightest: colors.swamp
};

export const createTheme = ({ palette }: ThemeOptions = {}) => {
  return createMuiTheme({
    palette: {
      primary: {
        light: "#f1a4ff",
        main: colors.liliana,
        dark: "#8844cb",
        contrastText: colors.gideon
      },
      secondary: {
        light: "#876dff",
        main: colors.island,
        dark: "#0015b4",
        contrastText: colors.gideon
      },
      common: colors,
      background: {
        ...backgrounds
      },
      ...palette
    },
    typography: {
      // Use the system font instead of the default Roboto font.
      fontFamily: [
        "Roboto",
        "Helvetica Neue",
        "Segoe UI",
        "-apple-system",
        "BlinkMacSystemFont",
        "Arial",
        "sans-serif",
        "Apple Color Emoji",
        "Segoe UI Emoji",
        "Segoe UI Symbol"
      ].join(",")
    },
    dimensions: {
      widths: {
        desktop: 1020,
        tablet: 720,
        mobile: 420
      }
    },
    overrides: {
      MuiPaper: {
        root: {
          backgroundColor: colors.swamp
        }
      },
      MuiTypography: {
        root: {
          letterSpacing: "0.024rem"
        }
      },
      MuiButton: {
        root: {
          minHeight: 33,
          textTransform: "initial"
        },
        label: {
          color: colors.ajani
        }
      },
      MuiInput: {
        root: {
          border: "2px solid transparent",
          transition: "border-color 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          borderRadius: 4,
          color: colors.karn,
          backgroundColor: backgrounds.darkest,
          "&:before, &:after": {
            display: "none"
          },
          "&:hover": {
            borderColor: colors.ral,

            "& $input, & > svg": {
              color: colors.ajani
            }
          },
          "&:hover&:focus-within, &:focus-within": {
            borderColor: colors.liliana,

            "& $input, & > svg": {
              color: colors.gideon
            }
          }
        },
        input: {
          padding: "13px 18px",
          fontSize: "14px"
        }
      },
      MuiAppBar: {
        root: {
          boxShadow: "none",
          backgroundColor: backgrounds.dark
        }
      },
      MuiToolbar: {
        regular: {
          maxHeight: "50px !important",
          minHeight: "50px !important"
        }
      },
      MuiSnackbarContent: {
        root: {
          backgroundColor: colors.liliana
        },
        message: {
          color: colors.gideon
        }
      },
      MuiTooltip: {
        tooltip: {
          backgroundColor: colors.ral
        }
      }
    },
    spacing: 9
  });
};
