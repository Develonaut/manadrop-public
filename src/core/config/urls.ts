import { compile } from "path-to-regexp";
import queryString from "query-string";
import urljoin from "url-join";

export const Urls = {
  COLLECTIONS: "/collections"
};

export const APIUrls = {
  SCRYFALL_CARD_SEARCH: "//api.scryfall.com/cards/search",
  SCRYFALL_CARD_LOOKUP: "//api.scryfall.com/cards/:id",
  SCRYFALL_CARD_NAME: "//api.scryfall.com/cards/named"
};

interface GetUrl {
  url: string;
  pathArgs?: object;
  queryParams?: object;
  encode?: boolean;
}

export function getUrl({
  url,
  pathArgs = {},
  queryParams = {},
  encode = false
}: GetUrl): string {
  const compiledUrl = compile(url)(pathArgs);
  const serializedParams = queryString.stringify(queryParams, { encode });
  const delimeter = url.includes("?") ? "&" : "?";
  return serializedParams
    ? urljoin(`${compiledUrl}${delimeter}${serializedParams}`)
    : urljoin(compiledUrl);
}
