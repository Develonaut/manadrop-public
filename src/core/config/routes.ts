import { IconDefinition } from "@fortawesome/fontawesome-common-types";
import { getUrl, Urls } from "core/config/urls";
import { ComponentType } from "react";

import { faThLarge } from "@fortawesome/free-solid-svg-icons";
import { CollectionsNav } from "features/subNavs/collections/CollectionsNav";
import { Collection } from "pages/Collection";

export interface RouteComponentProps {
  className?: string;
}

export interface RouteComponent {
  Component: ComponentType<RouteComponentProps>;
  props: object;
}

export interface Nav extends RouteComponent {
  Icon: IconDefinition;
}

export interface Routes {
  key: string;
  name: string;
  path: string;
  to: string;
  page: RouteComponent;
  nav: Nav;
}

export const routes = [
  {
    key: "collections",
    name: "Collections",
    nav: {
      Component: CollectionsNav,
      Icon: faThLarge,
      props: {}
    },
    path: Urls.COLLECTIONS,
    to: getUrl({ url: Urls.COLLECTIONS }),
    page: {
      Component: Collection,
      props: {}
    }
  }
];
