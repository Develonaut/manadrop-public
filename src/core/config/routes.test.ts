import { routes } from "core/config/routes";

describe("routes", () => {
  test("should export an array of objects containg route props", () => {
    expect(routes).toMatchSnapshot();
  });
});
