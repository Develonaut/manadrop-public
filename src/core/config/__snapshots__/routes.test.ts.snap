// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes should export an array of objects containg route props 1`] = `
Array [
  Object {
    "key": "collections",
    "name": "Collections",
    "nav": Object {
      "Component": [Function],
      "Icon": Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f009",
          "M296 32h192c13.255 0 24 10.745 24 24v160c0 13.255-10.745 24-24 24H296c-13.255 0-24-10.745-24-24V56c0-13.255 10.745-24 24-24zm-80 0H24C10.745 32 0 42.745 0 56v160c0 13.255 10.745 24 24 24h192c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24zM0 296v160c0 13.255 10.745 24 24 24h192c13.255 0 24-10.745 24-24V296c0-13.255-10.745-24-24-24H24c-13.255 0-24 10.745-24 24zm296 184h192c13.255 0 24-10.745 24-24V296c0-13.255-10.745-24-24-24H296c-13.255 0-24 10.745-24 24v160c0 13.255 10.745 24 24 24z",
        ],
        "iconName": "th-large",
        "prefix": "fas",
      },
      "props": Object {},
    },
    "page": Object {
      "Component": Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(CollectionPresentation)",
        "type": [Function],
      },
      "props": Object {},
    },
    "path": "/collections",
    "to": "/collections",
  },
]
`;
